### ISSUE TITLE
[bug]: Multiple listeners are alive (socket.io ) if connect and disconnect clicked many times.

### ISSUE THREAD

**Issue by sahell0x** [sparrowapp-dev/sparrow-app#2140]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

[Screencast from 2024-12-02 15-23-59.webm](https://github.com/user-attachments/assets/a360b1c2-3fe5-4768-a599-0ba9b32dbf5c)

### Steps to reproduce

go to socket.io connect and disconnect to any socket server you will see numerous events are there for single message

### Operating System

Others

**Comment by sahell0x** [comment]
@LordNayan  i am working on this

**Comment by LordNayan** [comment]
Sure @sahell0x please go ahead.

**Comment by sahell0x** [comment]
@LordNayan yes, i am synchronized with the upstream. it seems this could be a system dependent issue. i reviewed the UI, and it appears to be functioning perfectly. the problem seems to be with the socket.io client causing listener issues. since you are unable to reproduce the issue on your machine, it might no longer be a concern especially as sparrow has not  launched natively for linux.
